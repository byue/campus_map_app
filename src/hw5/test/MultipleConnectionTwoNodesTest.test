# Tests that a linear graph of 2 nodes can be created,
# and that the two nodes can have multiple connections
# with different labels.

# Create a graph
CreateGraph graph1

# Add 5 nodes
AddNode graph1 n1
AddNode graph1 n2

# Connect nodes with multiple connections
AddEdge graph1 n1 n2 a
AddEdge graph1 n2 n1 a
AddEdge graph1 n1 n2 c
AddEdge graph1 n2 n1 c
AddEdge graph1 n1 n2 b
AddEdge graph1 n2 n1 b

# list nodes of graph
ListNodes graph1

# list children of each node
ListChildren graph1 n1
ListChildren graph1 n2