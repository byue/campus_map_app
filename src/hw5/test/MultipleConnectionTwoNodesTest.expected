# Tests that a linear graph of 2 nodes can be created,
# and that the two nodes can have multiple connections
# with different labels.

# Create a graph
created graph graph1

# Add 5 nodes
added node n1 to graph1
added node n2 to graph1

# Connect nodes with multiple connections
added edge a from n1 to n2 in graph1
added edge a from n2 to n1 in graph1
added edge c from n1 to n2 in graph1
added edge c from n2 to n1 in graph1
added edge b from n1 to n2 in graph1
added edge b from n2 to n1 in graph1

# list nodes of graph
graph1 contains: n1 n2

# list children of each node
the children of n1 in graph1 are: n2(a) n2(b) n2(c)
the children of n2 in graph1 are: n1(a) n1(b) n1(c)